src/Components/GameBoard.js

```diff
@@ -577,6 +609,16 @@ class GameBoard extends Component {
       card_3 = this.state.board['cards'][this.state.board['turn']][2]
     }
 
+    if (!this.state.game_over) {
```

Comment:
Add comments to the front-end code. Note that due to the existing bugs this is not stable so need to fix all the front-end bugs going forward without introducing tech depth which new features have difficult time.